pkgs/by-name/README.md

```diff
@@ -76,6 +76,17 @@ and override its value in [`pkgs/top-level/all-packages.nix`](../top-level/all-p
 }
 ```
 
+### Toplevel argument restrictions
+The list of toplevel arguments should be restricted to packages defined in the
```

Comment:
Well, technically nothing hinders someone of creating a package with the exact same name of a Boolean flag.

What about with-shell?
https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/applications/misc/with-shell/default.nix

Reply:
I am saying that `with-shell` is an existing package, and because of `callPackage` no one can use `with-shell` as a simple Boolean flag.